src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
         /// </remarks>
-        public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
I have not seen anything in any of the tuple specifications that hint that there could be any choice in the underlying type. A tuple type is defined as equivalent to a specific underlying type, and that is the one consumers of the preferred API should get.